src/site/stages/build/tome-sync-page-builder.js

```diff
@@ -0,0 +1,159 @@
+  // on individual node / entity types, we can remove these from here.
+  // See the jsdoc on getEntityProperties for more information.
+  'field_facility_location',
+  'field_regional_health_service',
+  'field_region_page',
+  'field_office',
+]);
+
+/**
+ * Use to consistently reference to an entity.
+ *
+ * @param {String} type - The type of entity; corresponds to the file
+ *                        name.
+ * @param {String} uui  - The uuid of the entity; corresponds to the
+ *                        file name.
+ */
+const id = (type, uuid) => `${type}.${uuid}`;
+
+/**
+ * Take the entityName, return the contents of the file.
+ *
+ * Later, we can keep a counter for how many times we open a
+ * particular file to see if we can gain anything from caching the
+ * contents.
+ * @param {String} entityType - The type of entity; corresponds to the
+ *                              name of the file.
+ * @param {String} uuid - The UUID of the entity; corresponds to the
+ *                        name of the file.
+ */
+const fetchEntity = (type, uuid) =>
+  JSON.parse(
+    fs
+      .readFileSync(path.join(contentDir, `${type}.${uuid}.json`))
+      .toString('utf8'),
+  );
+
+/**
+ * Takes the type of entity and entity itself, returns an array of
+ * properties that we want to keep for this entity.
+ *
+ * @param {String} entityType - The type of entity; corresponds to the
+ *                              name of the file.
+ * @param {Object} entity - The contents of the entity itself before
+ *                          reference expansion.
+ *
+ * @return {Array<String>} - A list of properties of the entity to
+ *                           keep.
+ */
+const getEntityProperties = (entityType, entity) =>
+  // TODO: Change this to check the entity type and subtype and return
+  // a customized list.
+  Object.keys(entity);
+
+/**
+ * Takes an entity type and uuid, reads the corresponding file,
+ * searches for references to other entities, and replaces the
+ * references with the contents of those entities recursively.
+ *
+ * @param {String} entityType - The type of entity; corresponds to the
+ *                              name of the file.
+ * @param {String} uuid - The UUID of the entity; corresponds to the
+ *                        name of the file.
+ * @param {Array<String>} parents - A list of parent entities; used to
+ *                                  avoid circular dependencies.
+ *
+ * @return {Object} - The entity with all the references filled in with
+ *                    the body of the referenced entities.
+ */
+const assembler = (entityType, uuid, parents = []) => {
+  // Avoid circular dependencies
+  if (parents.includes(id(entityType, uuid))) {
+    /* eslint-disable no-console */
+    console.log(`I'm my own grandpa! (${id(entityType, uuid)})`);
+    console.log(`  Parents:\n    ${parents.join('\n    ')}`);
+    /* eslint-enable no-console */
+
+    // If we find a circular dependency, it needs to be addressed;
```

Comment:
nit: These are called `entity references` in the Drupal CMS, fwiw. So using the term `circular references` may fit better.

Reply:
Ah, good point. :+1: